zproject/urls.py

```diff
@@ -547,6 +547,16 @@
         ),
         name="password_reset_confirm",
     ),
+    path(
+        "accounts/weak_password/reset/<uidb64>/<token>/",
+        PasswordResetConfirmView.as_view(
+            success_url="/accounts/password/done/",
+            template_name="zerver/reset_confirm.html",
+            form_class=LoggingSetPasswordForm,
+            extra_context={"weak_password": True},
+        ),
+        name="change_weak_password",
+    ),
```

Comment:
Why do we need a separate URL, rather than just an extra parameter `?weak_password=True`, for example?

Reply:
We cannot use GET parameters for `PasswordResetConfirmView` because of how it was written.

This is how it works along with the what's restricting us from using `?weak_password=True`:
* When the browser requests `accounts/weak_password/reset/<uidb64>/<token>/`, django saves the `token` in URL to session and then redirects to `accounts/weak_password/reset/<uidb64>/set-password`.*(this is where our GET params get lost as django uses `request.path` to get our orignal URL(`accounts/weak_password/reset/`), but `request.path` doesn't include GET params)*
* Now, our "set a new password" template is rendered. (This is where we need that `weak_password: True` to tell the user their password is weak). On entering a new password and sending a post request, django validates the `token` from session and then resets the user's password.